### PR TITLE
added ISAs to architecture detection

### PR DESCRIPTION
**Pull request by jstucke**
added some missing embedded ISAs to architecture detection plugin:
- ARC
- AVR
- RISC-V
- ESP

### CODE REVIEW

File: src/plugins/analysis/architecture_detection/code/architecture_detection.py [hunk 1]
```diff
@@ -59,21 +59,26 @@ class MetaDataDetector:
...
+        'ESP': ['Tensilica Xtensa'],
         'Alpha': ['Alpha'],
         'M68K': ['m68k', '68020']
     }
     bitness = {
+        '8-bit': ['8-bit'],
         '16-bit': ['16-bit'],
         '32-bit': ['32-bit', 'PE32'],
-        '64-bit': ['64-bit', 'aarch64', 'x86-64', '80860']
+        '64-bit': ['64-bit', 'aarch64', 'x86-64', 'MIPS32', 'MIPS64', '80860']
```
Comment (rhelmke) [inline src/plugins/analysis/architecture_detection/code/architecture_detection.py:81]: MIPS32 is 32-bit :-)

**Review by rhelmke — CHANGES_REQUESTED** [review]

File: src/plugins/analysis/architecture_detection/code/architecture_detection.py (hunk 1 above)
Comment (jstucke) [reply src/plugins/analysis/architecture_detection/code/architecture_detection.py:81]: whoops good catch :sweat_smile: -> moved to '32-bit'

File: src/plugins/analysis/architecture_detection/code/architecture_detection.py [hunk 2]
```diff
@@ -19,7 +19,7 @@ def __init__(self, plugin_administrator, config=None, recursive=True):
         default flags should be edited above. Otherwise the scheduler cannot overwrite them.
         '''
         self.config = config
-        self.MIME_BLACKLIST = [
+        self.MIME_BLACKLIST = [  # pylint: disable=invalid-name
```
[the remark this answers is not in the document]
Comment (jstucke) [reply src/plugins/analysis/architecture_detection/code/architecture_detection.py:22]: I'm not even sure why I didn't notice that -> converted to a class constant

**Review by rhelmke — APPROVED** [review]
Both tests and CI builds pass, looks good to me now :-)

**Pull request merged**